cmd/bucket-replication.go

```diff
@@ -690,23 +701,34 @@ func replicateObject(ctx context.Context, ri ReplicateObjectInfo, objectAPI Obje
 				if !ok {
 					return
 				}
+				gr.Close()
+				closeOnDefer = false
 				// This lower level implementation is necessary to avoid write locks from CopyObject.
 				poolIdx, err := z.getPoolIdx(ctx, bucket, object, objInfo.Size)
```

Comment:
No reason to use `getPoolIdx()` @poornas you can simply remove the entire reflection code and directly called `PutObjectMetadata()`